tests/TestCase.php

```diff
@@ -2,9 +2,35 @@
 
 namespace Tests;
 
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Database\Eloquent\Relations\HasMany;
 use Illuminate\Foundation\Testing\TestCase as BaseTestCase;
 
 abstract class TestCase extends BaseTestCase
 {
     use CreatesApplication;
+
+    protected function hasMany_relation_test($related, $foreignKey, $relation)
+    {
+        $this->assertInstanceOf(HasMany::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
+    protected function belongsTo_relation_test($related, $foreignKey, $ownerKey, $relation)
+    {
+        $this->assertInstanceOf(BelongsTo::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($ownerKey, $relation->getOwnerKeyName());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
+    protected function belongsToMany_relation_test($related, $foreignKey, $relatedKey, $relation)
+    {
```

Comment:
giải thich các function này

Reply:
e test relation của các model với nhau
$this->assertInstanceOf(BelongsTo::class, $relation); e test loại relation
$this->assertInstanceOf($related, $relation->getRelated()); test quan hệ model tương ứng
mấy đoạn sau là e test khóa ngoại với khóa chính
$this->assertEquals($ownerKey, $relation->getOwnerKeyName());
$this->assertEquals($foreignKey, $relation->getForeignKeyName());